docs/src/operate/lifecycle/deregister.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Deregister entities
+tags: [Lifecycle]
+sidebar_position: 1
+description: Deregister entities from thin-edge
+---
+
+# Deregister entities
+
+All the entities registered with thin-edge are stored with the MQTT broker as retained messages,
+with their metadata spread across multiple topics.
+For e.g, for a child device `child01`, the registration message is stored on the `te/device/child01//` topic,
+its twin data is stored across several `te/device/child01///twin/<twin-data-type>` topics,
+its command metadata stored across several `te/device/child01///cmd/<cmd-type>` topics and so on.
+
+On top of that, when a device has services associated with it, or has nested child devices,
+they all have their own respective registration topics and other metadata topics.
+So, de-registering a device involves de-registering itself, its metadata and
+the complete entity hierarchy that is associated with it.
+
+## Deregister a single entity
+
+De-registering a single entity, excluding its child entities, involves clearing its registration message and
+all the metadata messages associated with it from the MQTT broker.
+Even though thin-edge doesn't provide a direct API for this yet, this can be easily be done using third-party tools
+like `mosquitto_sub` that supports clearing multiple retained messages together using the `--remove-retained` option.
+
+For e.g, a child device with topic id `device/child01//` can be deregistered as follows:
+
+```sh
+mosquitto_sub --remove-retained -W 3 -t "te/device/child01///#"
+```
+
+The topic filter `te/device/child01///#` covers the device's registration message as well all of its metadata messages.
+Running the same command without the `remove-retained` option shows the list of messages that would be cleared with this command.
+The `-W 3` option is used to stop the command after a maximum timeout period of 3 seconds.
+Adjust this timeout value based on the number of entities in your deployment.
+
+:::note
+Deregistering the main device using this mechanism is not supported.
+:::
+
+Similarly, a service with topic id `device/main/service/service01` can be deregistered as follows:
+
+```sh
+mosquitto_sub --remove-retained -W 3 -t "te/device/main/service/service01/#"
+```
+
+Services of child devices can also be de-registered in a similar manner, using the child device topic id instead.
+
+:::note
+The Cumulocity mapper must be restarted for the de-registrations to take effect.
+The entities in the cloud must also be removed separately as de-registrations on the device
```

Comment:
```suggestion
The entities in the cloud must also be removed separately as deregistration on the device
```

Reply:
I'm curious, `de-registering` requires `-` whilst `deregistrations` doesn't?

Reply:
Yeah, it's weird. When I google the following variants without the `-` seemed valid:

1. deregister
2. deregistered
3. deregistering
4. deregistration

But the spellchecker plugin in our devcontainer was specifically unhappy with `deregistering` and `deregistration` while the other two were fine. But, it looks like the spell checker in our workflow is happy with the non-hyphen versions. So, I'll just use those everywhere to be consistent.